ice.c

```diff
@@ -1010,10 +1010,19 @@ int janus_ice_set_stun_server(gchar *stun_server, uint16_t stun_port) {
 
 	/* Test the STUN server */
 	janus_network_address public_addr = { 0 };
-	if(janus_ice_test_stun_server(&addr, janus_stun_port, 0, &public_addr, NULL) < 0)
+	if(janus_ice_test_stun_server(&addr, janus_stun_port, 0, &public_addr, NULL) < 0) {
+		if(janus_stun_server) {
```

Comment:
No need for the `if`, `g_free` does nothing if it's `NULL`.

Reply:
good to know :-)